Automattic/jetpack
# Backup: Update purchase flow to use unlinked=1

#### Changes proposed in this Pull Request:

* This uses the updated functionality of the Connection UI to pass a `siteProductAvailabilityHandler`

#### Jetpack product discussion


Some discussion here: https://github.com/Automattic/jetpack/issues/23171

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Apply this branch to a site (like Jurassic Ninja) via Jetpack Beta
* Ensure you can connect the site and purchase a plan (and your cart has `unlink=1` in the URL)
* Deactivate the Jetpack Backup plugin and reactivate (may need to through Jetpack Beta again)
* Reconnect the Jetpack Backup plugin and ensure you aren't brought to checkout.
* Try the same steps, but with a site with a legacy plan, to ensure you aren't put into checkout while already having a valid plan, like Jetpack Personal.
